### PR TITLE
App 13900 extend swirl navigation item

### PR DESCRIPTION
**Pull request by webspaceadam**
## Summary

- [Ticket](https://flipapp.myjetbrains.com/issue/APP-13900)
- [Design](https://www.figma.com/file/iRHiAFsHxnzux8gQ9EjlZe/Swirl-Components?node-id=3215%3A21685&t=89Ox7YPjoH5rRxN7-0)
- [Storybook App Icon](https://gentle-water-054ae5203-184.westeurope.1.azurestaticapps.net/?path=/docs/components-swirlappicon--swirl-app-icon) & [Shell Layout](https://gentle-water-054ae5203-184.westeurope.1.azurestaticapps.net/?path=/docs/components-swirlshelllayout--swirl-shell-layout)

### CODE REVIEW

File: packages/swirl-components/src/components/swirl-shell-layout/swirl-shell-layout.css [hunk 1]
```diff
@@ -134,22 +135,19 @@
   display: flex;
   overflow: hidden;
   align-items: center;
+  height: 4.5rem;
```
Comment (Sqrrl) [inline packages/swirl-components/src/components/swirl-shell-layout/swirl-shell-layout.css:138]: **issue:** Logo and button aren't centered.

Bildschirm­foto 2023-03-06 um 09 04 32

Comment (Sqrrl) [inline packages/swirl-components/src/components/swirl-shell-layout/swirl-shell-layout.css:138]: **issue:** Spacing to the navigation is off.
Bildschirm­foto 2023-03-06 um 09 06 44

File: packages/swirl-components/src/components/swirl-shell-navigation-item/swirl-shell-navigation-item.css [hunk 2]
```diff
@@ -50,19 +48,36 @@
...
+
 .shell-navigation-item__icon {
   display: inline-flex;
-
-  & ::part(icon) {
-    color: var(--s-icon-default);
-  }
+  justify-content: center;
+  align-items: center;
+  min-width: 40px;
```
Comment (Sqrrl) [inline packages/swirl-components/src/components/swirl-shell-navigation-item/swirl-shell-navigation-item.css:79]: **nitpick:** rem for better scaling with font size settings.

File: packages/swirl-components/src/components/swirl-shell-navigation-item/swirl-shell-navigation-item.css [hunk 3]
```diff
@@ -71,3 +86,25 @@
   white-space: nowrap;
   text-overflow: ellipsis;
 }
+
+.shell-navigation-item__badge {
+  margin-left: auto;
+}
+
+@container (max-width: 64px) {
```
Comment (Sqrrl) [inline packages/swirl-components/src/components/swirl-shell-navigation-item/swirl-shell-navigation-item.css:94]: **nitpick:** See above

File: packages/swirl-components/src/components/swirl-shell-navigation-item/swirl-shell-navigation-item.css [hunk 4]
```diff
@@ -71,3 +86,25 @@
...
+  .shell-navigation-item__label {
+    display: none;
+  }
+
+  .shell-navigation-item__badge {
+    position: absolute;
+    bottom: 0;
+    right: 0.25rem;
```
Comment (Sqrrl) [inline packages/swirl-components/src/components/swirl-shell-navigation-item/swirl-shell-navigation-item.css:102]: **nitpick:** `var(--s-space-4)`

File: packages/swirl-components/src/components/swirl-shell-navigation-item/swirl-shell-navigation-item.css [hunk 5]
```diff
@@ -71,3 +86,25 @@
...
+  }
+}
+
+@container (min-width: 269px) {
```
Comment (Sqrrl) [inline packages/swirl-components/src/components/swirl-shell-navigation-item/swirl-shell-navigation-item.css:106]: **nitpick:** See above (rem).

File: packages/swirl-components/src/components/swirl-shell-navigation-item/swirl-shell-navigation-item.tsx [hunk 6]
```diff
@@ -49,12 +35,18 @@ export class SwirlShellNavigationItem {
...
-        ></span>
+        <span class="shell-navigation-item__icon">
+          <slot name="icon"></slot>
+        </span>
         <span class="shell-navigation-item__label">{this.label}</span>
+        {this.badgeLabel && (
+          <swirl-badge
+            class="shell-navigation-item__badge"
+            size="s"
+            aria-label={`${this.badgeLabel} 1 new messages`}
```
Comment (Sqrrl) [inline packages/swirl-components/src/components/swirl-shell-navigation-item/swirl-shell-navigation-item.tsx:46]: **issue:** Remove hard coded `1 new messages`

**Review by Sqrrl — CHANGES_REQUESTED** [review]

File: packages/swirl-components/src/components/swirl-app-icon/swirl-app-icon.css [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    box-sizing: border-box;
+  }
+}
+
+.app-icon {
+  position: relative;
+  overflow: hidden;
+  width: 2.5rem;
+  height: 2.5rem;
+  max-width: 2.5rem;
```
[the remark this answers is not in the document]
Comment (webspaceadam) [reply packages/swirl-components/src/components/swirl-app-icon/swirl-app-icon.css:14]: and what happens, when someone adds a really weird image? should still be a square-like object through overflow, right?

[the remark this answers is not in the document]
Comment (Sqrrl) [reply packages/swirl-components/src/components/swirl-app-icon/swirl-app-icon.css:14]: The element would still be limited to the fixed width of `2.5rem`. I think the usage of a fixed `width` makes the `max-width` not do anything. Or am I missing something?

**Review by Sqrrl — APPROVED** [review]

**Pull request merged**